### PR TITLE
Session test

### PR DESCRIPTION
**Pull request by jburel**
Add test to check if the ```_del_``` method actually deletes the session
This is not the case with ice version < 3.6
cf. https://trac.openmicroscopy.org/ome/ticket/12921

I have fixed a bug in the disconnect method. We will need to watch if some tests failed (wrong assumption in the tests)

### CODE REVIEW

File: components/tools/OmeroJava/test/integration/AbstractServerTest.java [hunk 1]
```diff
@@ -698,8 +698,8 @@ protected void makeGroupOwner() throws Exception {
      */
     protected omero.client disconnect() throws Exception {
         omero.client oldClient = client;
-        client = null;
         clean();
+        client = null;
```
Comment (joshmoore) [inline components/tools/OmeroJava/test/integration/AbstractServerTest.java]: That's a nasty test bug. So is the reason why this wasn't needing in Ice 3.6 that Glacier is doing a better job of cleaning up objects than before?

Comment (jburel) [inline components/tools/OmeroJava/test/integration/AbstractServerTest.java]: The ``disconnect`` method was never invoking the ``del`` method nothing related to ice 3.6

Comment (jburel) [inline components/tools/OmeroJava/test/integration/AbstractServerTest.java]: as expected https://ci.openmicroscopy.org/view/OMERO-DEV/job/OMERO-DEV-merge-integration-java/255/

**Comment by jburel** [comment]
The tests should now turned green.

**Comment by jburel** [comment]
Adding breaking label so the PR can be tested tomorrow with both ice 3.5 and ice 3.6 on breaking. now that https://ci.openmicroscopy.org/view/Breaking/job/OMERO-DEV-breaking-build is green cf. https://ci.openmicroscopy.org/view/Breaking/job/OMERO-DEV-breaking-build/637/

File: components/tools/OmeroJava/test/integration/chgrp/HierarchyMoveDatasetTest.java [hunk 2]
```diff
@@ -106,7 +106,7 @@ private void moveDataDatasetToProject(String source, String target,
                 .simpleDatasetData().asIObject());
 
         // log out
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/chgrp/HierarchyMoveDatasetTest.java]: seems odd to leave the comment behind

**Comment by mtbc** [comment]
Tests all pass on breaking! Can't see from CI that both Ice 3.5 and Ice 3.6 are covered -- do we switch something and then look again tomorrow?

https://github.com/snoopycrimecop/openmicroscopy/compare/develop/breaking/trigger...snoopycrimecop:develop/merge/trigger shows a merge conflict but perhaps I'm looking at the wrong branches.

**Comment by mtbc** [comment]
(Ah, @sbesson tells me the develop one is from the devspace.)

**Comment by mtbc** [comment]
https://github.com/snoopycrimecop/openmicroscopy/compare/develop/breaking/trigger...snoopycrimecop:develop/merge/daily has the same issue.

**Comment by jburel** [comment]
@mtbc:

* integration tests with ice 3.6 http://10.0.51.107:8080/job/OMERO-test-integration/label=testice36/
* integration tests with ice 3.5 http://10.0.51.107:8080/job/OMERO-test-integration/label=testice35/

**Comment by mtbc** [comment]
OMERO-push includes,
```
# PR 4594 jburel 'Session test'
```

Failure of `OmeroPy.test.integration.clitest.test_fs.TestFS.testSetsWithTransfer` involves `Invalid cross-device link` which looks more like a CI setup issue. Seems to me to be good to merge.

**Comment by jburel** [comment]
@mtbc: that's the one I was mentioning this am, not related to ice but more set-up

**Comment by aleksandra-tarkowska** [comment]
@mtbc I have exactly the same issue across all running devspaces, not sure what permissions has to be change and where exactly

**Comment by mtbc** [comment]
Might be interesting to check which directories are being assumed to be on the same partition.

**Comment by aleksandra-tarkowska** [comment]
It turned out that is not a problem of directories or permissions, just lack of right ports build opened. CLI and Insight requires 4063 and 4064. devspace should be green (blue) tomorrow

**Pull request merged**